src/theory/quantifiers/quant_bound_inference.cpp

```diff
@@ -88,7 +88,13 @@ BoundVarType QuantifiersBoundInference::getBoundVarType(Node q, Node v)
 {
   if (d_bint)
   {
-    return d_bint->getBoundVarType(q, v);
+    BoundVarType bvt = d_bint->getBoundVarType(q, v);
+    // If the bounded integer module has a bound, use it. Otherwise, we fall
+    // through.
+    if (bvt != BOUND_NONE)
+    {
+      return bvt;
+    }
   }
```

Comment:
Can you add a test that reveals the bug that this solves?

Reply:
There are roughly 15 regressions that fail if this bug is not fixed.